cdap-security/src/main/java/io/cdap/cdap/security/tools/TokenGeneratorTool.java

```diff
@@ -0,0 +1,120 @@
+/*
+ * Copyright © 2022 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package io.cdap.cdap.security.tools;
+
+import ch.qos.logback.classic.Level;
+import ch.qos.logback.classic.Logger;
+import com.google.common.annotations.VisibleForTesting;
+import com.google.inject.Guice;
+import com.google.inject.Injector;
+import com.google.inject.Key;
+import com.google.inject.TypeLiteral;
+import io.cdap.cdap.api.common.Bytes;
+import io.cdap.cdap.common.conf.CConfiguration;
+import io.cdap.cdap.common.conf.Constants;
+import io.cdap.cdap.common.guice.ConfigModule;
+import io.cdap.cdap.common.guice.IOModule;
+import io.cdap.cdap.common.guice.InMemoryDiscoveryModule;
+import io.cdap.cdap.common.io.Codec;
+import io.cdap.cdap.security.auth.AbstractKeyManager;
+import io.cdap.cdap.security.auth.AccessToken;
+import io.cdap.cdap.security.auth.FileBasedKeyManager;
+import io.cdap.cdap.security.auth.TokenManager;
+import io.cdap.cdap.security.auth.UserIdentity;
+import io.cdap.cdap.security.guice.CoreSecurityRuntimeModule;
+import org.apache.commons.cli.BasicParser;
+import org.apache.commons.cli.CommandLine;
+import org.apache.commons.cli.CommandLineParser;
+import org.apache.commons.cli.HelpFormatter;
+import org.apache.commons.cli.Option;
+import org.apache.commons.cli.Options;
+import org.apache.commons.cli.ParseException;
+import org.slf4j.LoggerFactory;
+
+import java.io.IOException;
+import java.util.Base64;
+import java.util.Collections;
+
+/**
+ * A command line tool for generating access tokens.
+ */
+public class TokenGeneratorTool {
+
+  private void doMain(String[] args) throws ParseException, IOException {
+    Options options = new Options()
+      .addOption(new Option("h", "help", false, "Print this usage message."))
```

Comment:
Since this is meant to be a command-line tool which should be executable without relying on a mounted CConf, should we make the path to the key file configurable? Otherwise it will expect the key to be at a set file-path every time.

Reply:
The key manager depends on cConf, so I don't think we can eliminate the cConf dependency - https://github.com/cdapio/cdap/blob/0b061003a4d9eb7eeb26b2e52e943052baea8b28/cdap-security/src/main/java/io/cdap/cdap/security/auth/AbstractKeyManager.java#L62
FWIW AuthenticationTools also depends on cConf - https://github.com/cdapio/cdap/blob/develop/cdap-security/src/main/java/io/cdap/cdap/security/tools/AuthenticationTool.java#L82

Is there there a reason we don't want to introduce a dependency on cConf? The secret contain the key info needs to be mounted anyway, so why not also mount the cConf configmap?

Reply:
Will fix based on offline discussion

Reply:
fixed